osbrain/agent.py

```diff
@@ -898,6 +898,51 @@ def _connect_and_register(self, client_address, alias=None, handler=None,
         self.register(socket, register_as, alias, handler)
         return client_address
 
+    def unsubscribe_from_topic(self,
+                               socket: Union[AgentAddress, str, zmq.Socket],
+                               topic: bytes):
+        '''
+        Unsubscribe a socket from a given topic.
+
+        Parameters
+        ----------
+        socket
+            Identifier of the socket. Must be a valid entry of `self.socket`
+            for the PUB/SUB pattern and a valid entry of `self.address` for
+            the SYNC_PUB/SYNC_SUB pattern
+        topic
+            topic which we want to unsubscribe from
+        '''
+        if isinstance(self.address[socket], AgentAddress):
+            self.socket[socket].setsockopt(zmq.UNSUBSCRIBE, topic)
+        elif isinstance(self.address[socket], AgentChannel):
+            channel = self.address[socket]
+            socket = channel.receiver
+            treated_topic = channel.uuid + topic
+            self.socket[socket].setsockopt(zmq.UNSUBSCRIBE, treated_topic)
+
+    def subscribe_to_topic(self, socket: Union[AgentAddress, str, zmq.Socket],
+                           topic: bytes):
```

Comment:
We do allow `Union[bytes, str]` when making subscriptions (see Agent's `_subscribe` method). I think we should do the same here as well. User can directly use `str` (it will automatically be converted to bytes encoded in UTF8, but can use bytes if they prefer to.